service/create/service.go

```diff
@@ -492,10 +519,21 @@ func (s *Service) runMachine(input runMachineInput) error {
 			},
 		},
 	}); err != nil {
-		return microerror.MaskAny(err)
+		return nil, microerror.MaskAny(err)
 	}
 
 	s.logger.Log("info", fmt.Sprintf("instance '%s' tagged", input.name))
+	instance := reservation.Instances[0]
 
-	return nil
+	return instance, nil
```

Comment:
> I got the impression that we have a function to create resources which we now modify to additionally fetch information about the created resources in one step. 

The function returns  the instance we just created by [extracting](https://github.com/kinvolk/aws-operator/blob/befaafe424b5b4da4b3a46b8b1553480eeccc10b/service/create/service.go#L526) it from the [response](https://github.com/kinvolk/aws-operator/blob/befaafe424b5b4da4b3a46b8b1553480eeccc10b/service/create/service.go#L479) of the API call to [`ec2.runInstances`](https://docs.aws.amazon.com/sdk-for-go/api/service/ec2/#EC2.RunInstances). In that response we have all the data we need.

So we are not fetching anything in that method, only returning a portion of the API response.

Wouldn't it be inefficient to go and fetch that same information again, since we just got it?